.pipelines/azure_pipeline_dev.yaml

```diff
@@ -88,12 +97,24 @@ steps:
       az account set -s ${{ variables.subscription }}
       az acr login -n ${{ variables.containerRegistry }}
 
-      docker buildx build --platform linux/amd64,linux/arm64 --tag ${{ variables.repoImageName }}:$(datetime)-$(commit) -f kubernetes/linux/Dockerfile.multiarch --metadata-file $(Build.ArtifactStagingDirectory)/metadata.json --build-arg IMAGE_TAG=$(datetime)-$(commit) --push .
+      if [ "$(Build.Reason)" != "PullRequest" ]; then
```

Comment:
Few questions  1) in case of non-PullRequest(triggering the build), there is no --push, will that done by as part of  release step? 2) And because of this,  do we need to distinguish between PullRequest vs non-PullRequest

Reply:
1. with these changes PR and PR updates will build and trivy test them. These wont be pushed to the ACR. ACR push will happen when PR completes and is merged. 
2. Release should be done from a CI build once PR has merged and not a PR build.